Don't fail on missing sensu subscription

paramite opened the pull request:
Currently SensuConnector fails to connect when all subscriptions are not created
by Sensu server. This patch makes the connector fail only when connection failed
for all subscriptions.

paramite commented:
@pleimer Seems like you accidentally submitted PR to my personal repo. Created the PR here instead. Please resolve conflicts, so we can merge it.

leifmadsen commented:
I guessed and resolved the conflict :)

paramite commented on logging/logger.go:
```diff
@@ -16,62 +18,72 @@ const (
+	switch strings.ToLower(target) {
+	case "console":
+		logger.write = func(message string) error {
+			fmt.Print(message)
```
Println maybe? Input logs don't contain new line IIRC, so this will make the whole program run output single line of logs.

pleimer replied:
writeRecord() adds the newline here: https://github.com/infrawatch/apputils/blob/f13cb337a60cd8f74d0747a333f993716a3fc705/logging/logger.go#L128

paramite replied:
Ah, right. Then just ignore my comment :).

paramite reviewed (commented):
Except the writeline issue I am fine with this patch. Question is if this is not a bit over-engineering though. Pointing to stdout or stderr (as we do within Lokean [1]) does the same job. But anyway, I don't mind merging this once we don't write all logs to single line.

[1] https://github.com/paramite/collectd-sensubility/pull/4

paramite commented on logging/logger.go:
```diff
@@ -16,62 +18,72 @@ const (
-	self.Level = level
-}
-
-func(self *Logger) SetFile(path string, permissions os.FileMode) error {
```
Oh yes, almost missed that. Please don't remove this method as it is used in at least one project already.

pleimer replied:
nice catch. May have accidentaly deleted it

pleimer commented:
> Except the writeline issue I am fine with this patch. Question is if this is not a bit over-engineering though. Pointing to stdout or stderr (as we do within Lokean [1]) does the same job. But anyway, I don't mind merging this once we don't write all logs to single line.
> 
> [1] https://github.com/infrawatch/lokean/blob/master/cmd/main.go#L39

That's fair I hadn't thought of doing that when I originally made this patch to use in ci-server-go. If you want to stick with that, that works too.

pleimer commented:
I guess the only benefit to doing it this way is that it works on windowsl, though it may be as easy as pointing to some file stream there as well

paramite committed:
Don't fail on missing sensu subscription

Currently SensuConnector fails to connect when all subscriptions are not created
by Sensu server. This patch makes the connector fail only when connection failed
for all subscriptions.
